File: app/src/main/java/Views/UserCardList.java
package Views;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.lpiem.magiccards.R;

import java.util.ArrayList;

import controllers.InterfaceCallBackController;
import controllers.MagicCardRetrofitController;

public class UserCardList extends AppCompatActivity implements InterfaceCallBackController<Boolean> {
    ArrayList<String> listName = new ArrayList<>();
    TextView tv;
    private ArrayAdapter<String>  adapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_card_list);
        ListView listView = findViewById(R.id.listView);
        adapter = new ArrayAdapter<String>(UserCardList.this, android.R.layout.simple_list_item_1,listName);
        listView.setAdapter(adapter);

        MagicCardRetrofitController controller = new MagicCardRetrofitController((InterfaceCallBackController)this, listName);
        controller.callWS();
    }

    @Override
    public void onWorkDone(Boolean result) {
        if (result) adapter.notifyDataSetChanged();
        else Toast.makeText(UserCardList.this, "probleme de retour de l'API", Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/Views/UserCardList.kt
package Views

import Models.Card
import Views.Adapters.CardListAdapter
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import com.example.lpiem.magiccards.R
import controllers.InterfaceCallBackController
import controllers.MagicCardRetrofitController
import kotlinx.android.synthetic.main.activity_user_card_list.*

class UserCardList : AppCompatActivity(), InterfaceCallBackController {
    val animals: ArrayList<String> = ArrayList()
    var listCard = ArrayList<Card>()
    override fun onWorkDone(result: Any) {
        if (result as Boolean){
            rv_animal_list.adapter = CardListAdapter(listCard, this);
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user_card_list)

        val controller = MagicCardRetrofitController(this as InterfaceCallBackController)

        rv_animal_list.layoutManager = LinearLayoutManager(this);
        controller.callWS(listCard)
    }


}




